### PR TITLE
[video_player] duration updates event

### PR DESCRIPTION
**Pull request by zezo357**
I am the maintainer of the package [video_player_media_kit](https://pub.dev/packages/video_player_media_kit). While working with live streams, I encountered an issue where I needed to update the duration dynamically. However, I realized that there was no existing mechanism to update the duration.

This pull request aims to address this problem by introducing a new event called durationUpdate. This event allows the duration to be updated independently, providing greater flexibility when working with live streams. With this addition, developers will have the ability to send duration updates as needed.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by zezo357** [comment]
https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changing-federated-plugins following this i should get an approval on this first, then i should open a pr with the changes in the platform interface alone,

i did add the required tests in video_player, is there something else i should do, or only the approval remaining?

File: packages/video_player/video_player/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ## NEXT
 
 * Updates minimum supported SDK version to Flutter 3.3/Dart 2.18.
+* Adds listener to `VideoEventType.durationUpdated`.
```
Comment (hellohuanlin) [inline packages/video_player/video_player/CHANGELOG.md:4]: this needs a version bump so it can be released

**Review by hellohuanlin — COMMENTED** [review]
is this PR complete? I do not see relevant logic on native side that listens to duration change event.

**Comment by hellohuanlin** [comment]
can you provide more details on when the duration will change? is it specific to your package? or is it general behavior that this package may use in the future? is there other workarounds (such as using your own interface?)

**Comment by stuartmorgan** [comment]
> The duration changes when playing a live stream m3u8, while playing the duration will keep increasing

Please file an issue (you've checked the box, but didn't actually file an issue) with a reproducible test case, so that we can address this:

> Does exo player or iOS native use it and it's ignored in this package, I don't know sadly

There's no inherent issue with us adding a new API that is only used by some implementations, but I'd like to understand if that's actually the case here, and if so why. In at least some cases, both AVPlayer and ExoPlayer will report an unknown time value for live streams, and we should determine:
- If that's true for the cases you are trying to address in your implementation, or if they are also getting dynamic duration updates that we should be handling using this new API, and
- If they aren't, why `media_kit` is doing something different, and if those differences are actually meaningful, or if they should be elided for consistency. (E.g., if it's just always saying the duration is the current playback point or something like that, then that's probably not valuable enough to create new API to preserve it.)

**Comment by zezo357** [comment]
> > The duration changes when playing a live stream m3u8, while playing the duration will keep increasing
> 
> Please file an issue (you've checked the box, but didn't actually file an issue) with a reproducible test case, so that we can address this:
> 
> > Does exo player or iOS native use it and it's ignored in this package, I don't know sadly
> 
> There's no inherent issue with us adding a new API that is only used by some implementations, but I'd like to understand if that's actually the case here, and if so why. In at least some cases, both AVPlayer and ExoPlayer will report an unknown time value for live streams, and we should determine:
> - If that's true for the cases you are trying to address in your implementation, or if they are also getting dynamic duration updates that we should be handling using this new API, and
> - If they aren't, why `media_kit` is doing something different, and if those differences are actually meaningful, or if they should be elided for consistency. (E.g., if it's just always saying the duration is the current playback point or something like that, then that's probably not valuable enough to create new API to preserve it.)

The issue is related to my package should I file an issue when it's not a video_player issue?

Media_kit does the duration thing to allow people to seeking back when playing live videos 

I may check exoplayer and avplayer if they allow this type of implemention and comment again for this details

**Comment by stuartmorgan** [comment]
> The issue is related to my package should I file an issue when it's not a video_player issue?

You are trying to add functionality to `video_player`, by opening a PR against `video_player`; how is that not a `video_player` issue?

**Comment by zezo357** [comment]
> > The issue is related to my package should I file an issue when it's not a video_player issue?
> 
> You are trying to add functionality to `video_player`, by opening a PR against `video_player`; how is that not a `video_player` issue?
> 
> 

I am adding a feature in video_player that will help me fix an issue in my platform interface



For video_player it's a feature

For me it's an issue

**Comment by stuartmorgan** [comment]
The Flutter issue tracker (like most issue trackers) tracks both bug reports and feature requests, so I'm not sure why you are concerned about the distinction.

**Comment by zezo357** [comment]
> The Flutter issue tracker (like most issue trackers) tracks both bug reports and feature requests, so I'm not sure why you are concerned about the distinction.

In that case, I will open an issue explaining that playing live streams don't provide the duration updates, which is limiting the functionality of seeking back when playing

I guess that is a good general issue

**Comment by stuartmorgan** [comment]
@zezo357 Are you still planning on opening an issue for this to continue discussion?

**Comment by zezo357** [comment]
> @zezo357 Are you still planning on opening an issue for this to continue discussion?

Yes I am busy these days,  I found out exo player has this behavior

Of increasing the duration for live streams,

But didn't check iOS

**Comment by zezo357** [comment]
I created an issue

**Comment by camsim99** [comment]
Thank you for your contribution. It looks like the tests are failing. Before we review the PR, please see what you can do to resolve the test failures. If you are unsure how to proceed, please reach out for help on the #hackers-new channel.

**Comment by stuartmorgan** [comment]
I'm going to mark this as a draft for now while the question of what other platforms would do with this is resolved in the issue.

**Pull request closed without merging**